department-of-veterans-affairs/caseflow
# Production Backfill for HearingTasks

Resolves #9206 

### Description
Create HearingTasks for legacy appeals with ScheduleHearingTasks. Make the ScheduleHearingTask and HoldHearingTask, if it exists, children of the newly created HearingTask. Associate HearingTask with the ScheduleHearingTask original parent.

### NB: 
Although ticket mentions tasks for AMA appeals this part was taken care of by Alex's PR (see below). This PR focuses on LegacyAppeals with ScheduleHearingTasks that have RootTask as parent (and HoldHearingTasks if they exist).
A production query shows that there are 696 SHT and 77 HHT with RootTask as parent.

### Acceptance Criteria

### Testing Plan
1. Go to console.
2. Execute command: bundle exec rails runner scripts/back_fill_hearing_tasks.rb
3. With existing seed data confirm console message 'Total ScheduleHearingTasks to be relinked: 3' displays.
4. Verify Tasks with appeals 5, 6 and 7 were properly moved from RootTask as parent to HearingTask as parent.
